tests/PHPStan/Analyser/nsrt/bug-11561.php

```diff
@@ -0,0 +1,40 @@
+<?php // lint >= 8.0
+
+namespace Bug11561;
+
+use function PHPStan\Testing\assertType;
+use DateTime;
+
+/** @param array{date: DateTime} $c */
+function main(mixed $c): void{
+	assertType('array{date: DateTime}', $c);
+	$c['id']=1;
+	assertType('array{date: DateTime, id: 1}', $c);
+
+	$x = (function() use (&$c) {
+		assertType("array{date: DateTime, id: 1, name?: 'ruud'}", $c);
```

Comment:
I believe this is wrong, see https://github.com/phpstan/phpstan/issues/11945

Reply:
My reasoning is, that since this function is invoked immediately, it should be treated as the same code as normally, without the function.